samples/nrf9160/modem_shell/src/link/link_shell.c

```diff
@@ -1741,6 +1743,1441 @@ int link_shell(const struct shell *shell, size_t argc, char **argv)
+				mosh_error(
+					"APN string length %d exceeded. Maximum is %d.",
+					apn_len, LINK_APN_STR_MAX_LENGTH);
+				return -EINVAL;
+			}
+			apn = optarg;
+			break;
+		case 'f': /* Address family */
+			family = optarg;
+			break;
+		default:
+			break;
+		}
+	}
+
+	if (common_option == LINK_COMMON_READ) {
+		link_sett_defcont_conf_shell_print();
+	} else if (common_option == LINK_COMMON_ENABLE) {
+		link_sett_save_defcont_enabled(true);
+	} else if (common_option == LINK_COMMON_DISABLE) {
+		if (nrf_modem_at_printf("AT+CGDCONT=0") != 0) {
+			mosh_warn(
+				"ERROR from modem. Getting the initial PDP context back "
+				"wasn't successful.");
+		}
+		link_sett_save_defcont_enabled(false);
+		mosh_print("Custom default context config disabled.");
+	} else if (apn == NULL && family == NULL) {
+		link_shell_print_usage(LINK_CMD_DEFCONT);
+	}
+
+	if (apn != NULL) {
+		(void)link_sett_save_defcont_apn(apn);
+	}
+	if (family != NULL) {
+		enum pdn_fam pdn_lib_fam;
+
+		ret = link_family_str_to_pdn_lib_family(&pdn_lib_fam, family);
+		if (ret) {
+			mosh_error("Unknown PDN family %s", family);
+			link_shell_print_usage(LINK_CMD_DEFCONT);
+		} else {
+			(void)link_sett_save_defcont_pdn_family(pdn_lib_fam);
+		}
+	}
+
+	return 0;
+}
+
+static int link_shell_defcontauth(const struct shell *shell, size_t argc, char **argv)
+{
+	enum link_shell_common_options common_option = LINK_COMMON_NONE;
+	int protocol = 0;
+	bool protocol_given = false;
+	char *username = NULL;
+	char *password = NULL;
+
+	int long_index = 0;
+	int opt;
+
+	optreset = 1;
+	optind = 1;
+	while ((opt = getopt_long(argc, argv, short_options, long_options, &long_index)) != -1) {
+		link_shell_getopt_common_option(opt, &common_option);
+
+		switch (opt) {
+		case 'A': /* auth protocol */
+			protocol = atoi(optarg);
+			protocol_given = true;
+			break;
+		case 'U': /* auth username */
+			username = optarg;
+			break;
+		case 'P': /* auth password */
+			password = optarg;
+			break;
+		default:
+			break;
+		}
+	}
+
+	if (common_option == LINK_COMMON_READ) {
+		link_sett_defcontauth_conf_shell_print();
+	} else if (common_option == LINK_COMMON_ENABLE) {
+		if (link_sett_save_defcontauth_enabled(true) < 0) {
+			mosh_warn("Cannot enable authentication.");
+		}
+	} else if (common_option == LINK_COMMON_DISABLE) {
+		if (nrf_modem_at_printf("AT+CGAUTH=0,0") != 0) {
+			mosh_warn("Disabling of auth cannot be done to modem.");
+		}
+		link_sett_save_defcontauth_enabled(false);
+	} else if (!protocol_given && username == NULL && password == NULL) {
+
+		link_shell_print_usage(LINK_CMD_DEFCONTAUTH);
+	}
+
+	if (protocol_given) {
+		(void)link_sett_save_defcontauth_prot(protocol);
+	}
+	if (username != NULL) {
+		(void)link_sett_save_defcontauth_username(username);
+	}
+	if (password != NULL) {
+		(void)link_sett_save_defcontauth_password(password);
+	}
+
+	return 0;
+}
+
+static int link_shell_disconnect(const struct shell *shell, size_t argc, char **argv)
```

Comment:
`link disconnect` without parameters seems to try to do something it's not supposed to do (see the errors at the end):
```
mosh:~$ link disconnect
Usage: link connect --apn <apn str> [--family <pdn family str>] [auth options]
Options:
  -a, --apn, [str]    Access Point Name
  -f, --family, [str] PDN family: 'ipv4v6', 'ipv4', 'ipv6', 'non-ip'
Optional authentication options:
  -U, --uname, [str]  Username
  -P, --pword, [str]  Password
  -A, --prot,  [int]  Authentication protocol: 0 (None), 1 (PAP), 2 (CHAP)

Usage: link disconnect -I <cid>
Options:
  -I, --cid, [int]    Use this option to disconnect specific PDN CID

pdn_deactivate() failed, err -8
pdn_ctx_destroy() failed, err -22
<wrn> pdn: Failed to activate PDN for CID 0, err 65536
```
I don't see similar behavior with `link old disconnect`.

Reply:
Yep, this was something I had fixed but seems it got to a stash while rebasing and playing with CI issues.